src/components/diff-view.jsx

```diff
@@ -54,19 +55,27 @@ export default class DiffView extends React.Component {
     // TODO: if we have multiple ways to render content from a single service
     // in the future (e.g. inline vs. side-by-side text), we need a better
     // way to ensure we use the correct rendering and avoid race conditions
-    switch (diff.diff_service) {
-    case changeDiffTypes[diffTypes.SIDE_BY_SIDE_RENDERED]:
+    switch (diffType) {
```

Comment:
I’m with you on the need to clean up this diff type situation. (Dan and I talked briefly about it before, but I failed to fix it when I cleaned up #82 and merged it, so… my fault, sorry.)

The problem here is that `props.diffType` will be temporarily out-of-sync with `state.diff` when a user changes the selected diff type. We don’t want to render, say, viewable HTML data using the text diff component, so we need to make sure we pick the right differ for the content at hand. We can:

1. Make some bigger changes to how `constants/diff-types.js` is structured (well, we should probably do that anyway)
2. Remove responsibility for loading diff data from this component and move it up to `ChangeView`, which has more information about what is going on and can probably better coordinate (although that class has gotten huge and nutty).

I think the approach you suggested in the PR description here sounds good, but I haven’t thought through it fully. We should work through it tomorrow in person.